internal/provider/providers/porkbun/provider.go

```diff
@@ -119,46 +119,76 @@ func (p *Provider) Update(ctx context.Context, client *http.Client, ip netip.Add
 		recordType = constants.AAAA
 	}
 	ipStr := ip.String()
-	recordIDs, err := p.getRecordIDs(ctx, client, recordType)
+	records, err := p.getRecords(ctx, client, recordType, p.owner)
 	if err != nil {
 		return netip.Addr{}, fmt.Errorf("getting record IDs: %w", err)
 	}
 
-	if len(recordIDs) == 0 {
-		// ALIAS record needs to be deleted to allow creating an A record.
-		err = p.deleteALIASRecordIfNeeded(ctx, client)
-		if err != nil {
-			return netip.Addr{}, fmt.Errorf("deleting ALIAS record if needed: %w", err)
-		}
-
-		err = p.createRecord(ctx, client, recordType, ipStr)
-		if err != nil {
-			return netip.Addr{}, fmt.Errorf("creating record: %w", err)
+	if len(records) > 0 {
```

Comment:
Reverted to `len(recordIDs) == 0` given the new function `deleteDefaultConflictingRecordsIfNeeded` not taking as many lines in the if block 😉 (to have less deltas)